pip/exceptions.py

```diff
@@ -237,3 +237,7 @@ def hash_then_or(hash_name):
                          self.gots[hash_name].hexdigest())
             prefix = '    or'
         return '\n'.join(lines)
+
+
+class UnsupportedPythonVersion(InstallationError):
+    """Unsupported python version (related to PEP 345 Requires-Python)."""
```

Comment:
"related to PEP 345 Requires-Python" sounds a bit clumsy. Maybe "according to Requires-Python package metadata"?